src/value_generators/src/DocumentGenerator.cpp

```diff
@@ -1841,6 +1935,36 @@ UniqueGenerator<bsoncxx::array::value> literalArrayGenerator(const Node& node,
     return std::make_unique<LiteralArrayGenerator>(std::move(entries));
 }
 
+/**
+ * @param node sequence node
+ * @return array generator with a uniqueness filter for the node
+ */
+UniqueGenerator<bsoncxx::array::value> distinctArrayGenerator(const Node& node,
+                                                              GeneratorArgs generatorArgs) {
+    try {
+        if (extractKnownParser(node["of"], generatorArgs, intParsers)) {
+            // known parser type
+            return std::make_unique<DistinctArrayGenerator<int64_t>>(node, generatorArgs);
+        }
+    } catch (const InvalidConversionException& e) {
+    }
+    try {
+        if (extractKnownParser(node["of"], generatorArgs, doubleParsers)) {
+            // known parser type
+            return std::make_unique<DistinctArrayGenerator<double>>(node, generatorArgs);
+        }
+    } catch (const InvalidConversionException& e) {
+    }
+    try {
+        if (extractKnownParser(node["of"], generatorArgs, stringParsers)) {
+            // known parser type
+            return std::make_unique<DistinctArrayGenerator<std::string>>(node, generatorArgs);
+        }
+    } catch (const InvalidConversionException& e) {
+    }
+    BOOST_THROW_EXCEPTION(InvalidValueGeneratorSyntax("Unsupported \"of\" for distinct array"));
```

Comment:
Could we include the `of` node in the exception? Possibly even the whole node? Genny parsing error messages are generally awful, so any insight we can add is really helpful. This is a good start. Adding more information about the particular failing node would make it better.